api-authorization/server/lib/authorization-middleware.ts

```diff
@@ -0,0 +1,42 @@
+/* eslint-disable @typescript-eslint/no-unused-vars -- Remove me */
+import { NextFunction, Request, Response } from 'express';
+import jwt from 'jsonwebtoken';
+import { ClientError } from './client-error.js';
+
+export function authMiddleware(
+  req: Request,
+  res: Response,
+  next: NextFunction
+) {
+  /* your code here */
+  try {
+    const authHeader = req.get('Authorization');
+    if (!authHeader) throw new ClientError(401, 'Authentication required');
+    const token = authHeader.split('Bearer ')[1];
+    if (!token) throw new ClientError(401, 'Authentication required');
+    const payload = jwt.verify(token, process.env.TOKEN_SECRET as string);
```

Comment:
Nice try on the TypeScript problem. The real problem is that `process.env.TOKEN_SECRET` might be undefined, and `jwt.verify()` does not allow `undefined` for its second argument.